### PR TITLE
Nav menu Bulma component

### PR DESCRIPTION
**Pull request by rkofman**
Implements a basic version of the [Bulma navbar](https://bulma.io/documentation/components/navbar/) / menu. The existing code had it implemented as static HTML (already using Bulma styles). Clicking on the "hamburger" menu didn't do anything. The new code does a few things:

1. Makes the "hamburger" menu actually clickable (on small screens)
2. Extracts the "NavBar" into its own components that takes `children` for rendering the included links.

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -11,6 +11,7 @@
     "sourceMap": true,
     "strict": true
   },
+  "types": ["svelte"],
```
Comment (rkofman) [inline tsconfig.json:14]: Helps `eslint` understand built-in svelte types.

File: src-ui/routes/+layout.svelte [hunk 2]
```diff
@@ -7,32 +8,10 @@
...
-      </a>
-    </div>
-    <div class="navbar-menu">
-      <div class="navbar-start">
-        <a class="navbar-item" href="/">Home</a>
-        <a class="navbar-item" href="/about">About</a>
-      </div>
-    </div>
-  </nav>
+  <NavBar>
```
Comment (SilverMaiden) [inline src-ui/routes/+layout.svelte:11]: We should add a highlight on what's currently selected (or something similar!)

Comment (SilverMaiden) [reply src-ui/routes/+layout.svelte:11]: (Doesn't have to be in this PR)

**Review by SilverMaiden — APPROVED** [review]
Yay! 🎉

**Pull request merged**